### DISCUSSION TITLE
Calculate node position relative to block

### DISCUSSION THREAD

**Discussion by 0xLDev** [xyflow/xyflow#2744] [Q&A]
Need help how I correctly calculate the position of nodes, I want the added nodes to appear in the block with class actions
Below I provide screenshots with a custom node, and add nodes
![1](https://user-images.githubusercontent.com/67437278/211753001-c8e09b81-43e8-4fb8-9fa1-21a09e1ade6b.jpg)
![2](https://user-images.githubusercontent.com/67437278/211753012-af46ca43-2b6a-4349-840b-ab3c9d1e4364.jpg)

**Comment by amintabrizi** [comment]
I think, If you want to have a button next to the node, you can use the Toolbar feature. You can use the link below.
http://reactflow.dev/docs/api/nodes/node-toolbar/
Also if you want to get correct position of node in the browser or screen you can look to this Sandbox:
https://codesandbox.io/s/priceless-hopper-mq5j75?file=/src/App.js
This Sandbox is a simple example that show how to get current position of the node.

**Reply by 0xLDev** [reply]
thx
